store/sqlstore/group_store.go

```diff
@@ -359,6 +359,25 @@ func (s *SqlGroupStore) Delete(groupID string) (*model.Group, error) {
 	return &group, nil
 }
 
+func (s *SqlGroupStore) Undelete(groupID string) (*model.Group, error) {
+	var group model.Group
+	if err := s.GetReplicaX().Get(&group, "SELECT * from UserGroups WHERE Id = ? AND DeleteAt != 0", groupID); err != nil {
```

Comment:
Is there an advantage to having this `SELECT` query or could we skip it and attempt the `UPDATE` directly?

Reply:
1/5 in favour of this being here. It's clarity of the error message (vs) some performance cost which would be very rare.

Reply:
Sounds good, yes you're right it's rare.